### PR TITLE
sinkv2,processor: replace table ID with Span

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: ref #7720 

### What is changed and how it works?

* Replace TableID with tablepb.Span in sink v2

#### Tests 

 - Unit test
 - Integration test

##### Will it cause performance regression or break compatibility?

No

##### Do you need to update user documentation, design documentation or monitoring documentation?

No

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by 3AceShowHand — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

File: pkg/config/debug.go [hunk 1]
```diff
@@ -61,6 +62,16 @@ func (c *DebugConfig) ValidateAndAdjust() error {
 	if err := c.Scheduler.ValidateAndAdjust(); err != nil {
 		return errors.Trace(err)
 	}
+	if c.Scheduler.RegionPerSpan != 0 {
+		if c.EnablePullBasedSink || !c.EnableNewSink {
```
Comment (hi-rustin) [inline pkg/config/debug.go:66]: The `debug.enable-pull-based-sink` default is true. Does this mean you want to implement it in the table pipeline first?

Comment (overvenus) [reply pkg/config/debug.go:66]: Yes, for now, I think table pipeline is more stable for testing this feature.

**Pull request merged**